Loop/Plugins/LoopPlugins.swift

```diff
@@ -102,13 +104,51 @@ class PluginManager {
             return AvailableDevice(identifier: identifier, localizedTitle: title)
         })
     }
+
+    func getServiceTypeByIdentifier(_ identifier: String) -> ServiceUI.Type? {
+        for bundle in pluginBundles {
+            if let name = bundle.object(forInfoDictionaryKey: LoopPluginBundleKey.serviceIdentifier.rawValue) as? String, name == identifier {
+                do {
+                    try bundle.loadAndReturnError()
+
+                    if let principalClass = bundle.principalClass as? NSObject.Type {
+
+                        if let plugin = principalClass.init() as? ServiceUIPlugin {
+                            return plugin.serviceType
+                        } else {
+                            fatalError("PrincipalClass does not conform to ServiceUIPlugin")
+                        }
+
+                    } else {
+                        fatalError("PrincipalClass not found")
+                    }
+                } catch let error {
+                    print(error)
+                }
+            }
+        }
+        return nil
+    }
+
+    var availableServices: [AvailableService] {
+        return pluginBundles.compactMap({ (bundle) -> AvailableService? in
+            guard let title = bundle.object(forInfoDictionaryKey: LoopPluginBundleKey.serviceDisplayName.rawValue) as? String,
+                let identifier = bundle.object(forInfoDictionaryKey: LoopPluginBundleKey.serviceIdentifier.rawValue) as? String else {
+                    return nil
+            }
+
+            return AvailableService(identifier: identifier, localizedTitle: title)
+        })
+    }
+
 }
 
 
 extension Bundle {
     var isLoopPlugin: Bool {
         return
             object(forInfoDictionaryKey: LoopPluginBundleKey.pumpManagerIdentifier.rawValue) as? String != nil ||
-            object(forInfoDictionaryKey: LoopPluginBundleKey.cgmManagerIdentifier.rawValue) as? String != nil
+            object(forInfoDictionaryKey: LoopPluginBundleKey.cgmManagerIdentifier.rawValue) as? String != nil ||
```

Comment:
I think having to open up *all* the framework's plist files is going to be felt as slow by users. My original idea was to name the plugins with a .loopplugin extension, but that doesn't get past app store rules.  I'm now thinking plugin bundles should either contain a special file that denotes them as plugins, or the plugin installer script generates a manifest.  Leaning toward the second idea. No changes needed here; just thinking out loud. :)

Reply:
:+1: for the second idea.